eZ/Bundle/EzPublishCoreBundle/EventListener/ContentDownloadRouteReferenceListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\EventListener;
+
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\Core\MVC\Symfony\Event\RouteReferenceGenerationEvent;
+use eZ\Publish\Core\MVC\Symfony\MVCEvents;
+use InvalidArgumentException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ContentDownloadRouteReferenceListener implements EventSubscriberInterface
+{
+    const DOWNLOAD_ROUTE = 'content_download';
+    const DOWNLOAD_ROUTE_INTERNAL = '_ezplatform_download';
+    const FIELD_IDENTIFIER_ARGUMENT_NAME = 'field_identifier';
+    const CONTENT_ARGUMENT_NAME = 'content';
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            MVCEvents::ROUTE_REFERENCE_GENERATION => 'onRouteReferenceGeneration'
+        );
+    }
+
+    /**
+     * @throws \InvalidArgumentException If the required arguments are not correct
+     */
+    public function onRouteReferenceGeneration( RouteReferenceGenerationEvent $event )
+    {
+        $routeReference = $event->getRouteReference();
+
+        if ( $routeReference->getRoute() != self::DOWNLOAD_ROUTE )
+        {
+            return;
+        }
+
+        if (
+            !$routeReference->has( self::CONTENT_ARGUMENT_NAME )
+            ||
+            !$routeReference->has( self::FIELD_IDENTIFIER_ARGUMENT_NAME )
+        )
+        {
+            throw new InvalidArgumentException(
+                sprintf(
+                    "A %s route reference requires the '%s' and '%s' parameters",
+                    self::DOWNLOAD_ROUTE,
+                    self::CONTENT_ARGUMENT_NAME,
+                    self::FIELD_IDENTIFIER_ARGUMENT_NAME
+                )
+            );
+        }
```

Comment:
To avoid all the argument and type checks, you might want to use [OptionsResolver](http://symfony.com/doc/current/components/options_resolver.html).